drivers/hmis_csv_twenty_twenty_six/spec/models/importer/custom_data_element_integration_spec.rb

```diff
@@ -0,0 +1,439 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'CustomDataElement Integration' do
+  let(:data_source) { create(:grda_warehouse_data_source) }
+  let(:importer_log) { create(:hmis_csv_importer_log, data_source: data_source) }
+  let(:definition) { HmisCsvTwentyTwentySix.custom_files_config.find_definition('CustomDataElement.csv') }
+
+  # Create test warehouse records to lookup
+  let!(:test_client) { create(:grda_warehouse_hud_client, data_source: data_source, PersonalID: 'CLIENT123') }
+  let!(:test_enrollment) { create(:grda_warehouse_hud_enrollment, data_source: data_source, EnrollmentID: 'ENROLL456', PersonalID: test_client.PersonalID) }
+
+  describe 'CustomDataElement.csv column mapping' do
+    let(:source_record) do
+      HmisCsvTwentyTwentySix::Importer::Custom::CustomDataElement.new(
+        CustomDataElementID: 'CDE001',
+        CustomDataElementDefinitionID: 'DEF123',
+        RecordType: 'Client',
+        RecordID: 'CLIENT123',
+        Value: 'test_value',
+        DataCollectionStage: 1,
+        InformationDate: Date.parse('2024-01-01'),
+        UserID: 'user123',
+        DateCreated: DateTime.parse('2024-01-01T10:00:00Z'),
+        DateUpdated: DateTime.parse('2024-01-01T11:00:00Z'),
+        DateDeleted: nil,
+        ExportID: 'export456',
+        data_source_id: data_source.id,
+        importer_log_id: importer_log.id,
+      )
+    end
+
+    let(:mapped_attributes) { {} }
+
+    it 'successfully applies all column mappings including record lookup' do
+      mapper = HmisCsvTwentyTwentySix::Importer::Custom::ColumnMapper.new(definition.columns)
+      mapped_attributes = mapper.map(source_record)
+
+      # Test the record type mapping
+      expect(mapped_attributes['owner_type']).to eq('GrdaWarehouse::Hud::Client')
+
+      # Test the record lookup mapping
+      expect(mapped_attributes['owner_id']).to eq(test_client.id)
+
+      # Test direct mappings - use more flexible date/time matching
+      expect(mapped_attributes['CustomDataElementID']).to eq('CDE001')
+      expect(mapped_attributes['CustomDataElementDefinitionID']).to eq('DEF123')
+      expect(mapped_attributes['value_string']).to eq('test_value')
+      expect(mapped_attributes['DataCollectionStage']).to eq(1)
+      expect(mapped_attributes['UserID']).to eq('user123')
+      expect(mapped_attributes['ExportID']).to eq('export456')
+      expect(mapped_attributes['DateDeleted']).to be_nil
+
+      # Test dates with more flexible matching - allow both Date and DateTime
+      expect(mapped_attributes['InformationDate']).to be_a_kind_of(Time) # ActiveRecord often converts to Time/DateTime
```

Comment:
this one fails for me even after recreating my db. I get a date, not a time. Maybe:

```suggestion
            expect(mapped_attributes['InformationDate']).to satisfy { |v| v.acts_like?(:date) }
```

Same issue with the assertion on line 402

Reply:
Hmm, maybe I hadn't pushed my changes for this one.  it now checks:
```
expect(mapped_attributes['InformationDate']).to be_a_kind_of(Date) # ActiveRecord often converts to Date
```

Which is working for me.

Reply:
yeah sorry this works for me now, I probably had an old version. Thanks